src/config.js

```diff
@@ -256,7 +257,7 @@ export default class Config {
     this.modulesFolder = opts.modulesFolder;
     this.globalFolder = opts.globalFolder || constants.GLOBAL_MODULE_DIRECTORY;
     this.linkFolder = opts.linkFolder || constants.LINK_REGISTRY_DIRECTORY;
-    this.production = !!opts.production;
+    this.production = !!opts.production && opts.production !== 'false';
```

Comment:
Could do something like this in one line?
```javascript
    this.production = !!this.getOption('production') ||
      (opts.production !== 'false' &&
      process.env.NODE_ENV === 'production' &&
      process.env.NPM_CONFIG_PRODUCTION !== 'false' &&
      process.env.YARN_PRODUCTION !== 'false') ||
      !!opts.production && opts.production !== 'false';
```
Or
```javascript
    if (opts.production === 'false') {
      this.production = false;
    } else if (this.getOption('production') ||
        process.env.NODE_ENV === 'production' &&
        process.env.NPM_CONFIG_PRODUCTION !== 'false' &&
        process.env.YARN_PRODUCTION !== 'false') {
      this.production = true
    } else {
      this.production = !!opts.production;
    }
```

I prefer the latter. It also means that `production=false` will override the `getOption('production')`, which is probably the desired behaviour.